hazelcast/src/main/java/com/hazelcast/instance/impl/MobyNames.java

```diff
@@ -993,27 +993,26 @@
             "zhukovsky"};
 
     static {
-        Arrays.sort(LEFT, MobyNames::compareRandomly);
-        Arrays.sort(RIGHT, MobyNames::compareRandomly);
+        Collections.shuffle(Arrays.asList(LEFT));
+        Collections.shuffle(Arrays.asList(RIGHT));
```

Comment:
this is actually will have the same result.
The problem is not in shuffling, it is in how we make a sequence of names. So, if we have two arrays with sizes N and M, we randomly choose index I1 which is [0:N], then all subsequent "random" names will have first half with value from that index I1, so the names will be I1_L4, I1_l1, I1L5 ... I1_LM where L1-LM - random indexes from array M.

This pattern is only visible during tests. Actual real world usage is one name per one shuffling.

Reply:
I realize that the implementation combines the elements from two arrays as if they were two digits, so the surname is the same for all numbers from 0 to 99. I tried changing that with code that involved the BIG_PRIME but reverted because this may be a desired feature. For example, in tests you'll start many HZ instances from the same JVM and they'll all get the same surname.

However, the change in shuffling won't give the same result because the original method just didn't shuffle the array properly. I keep seeing "albattani", "babbage" , "boyd" but never see "spense", "stallman" or "tesla". The original shuffling technique uses a broken comparator that fails to satisfy its contract. This has unpredictable results and can even result in infinite loops.